cirq-core/cirq/experiments/two_qubit_xeb.py

```diff
@@ -156,9 +157,109 @@ def plot_histogram(self, ax: Optional[plt.Axes] = None, **plot_kwargs) -> plt.Ax
             fig.show(**plot_kwargs)
         return ax
 
+    @cached_method
+    def pauli_error(self) -> Dict[Tuple['cirq.GridQubit', 'cirq.GridQubit'], float]:
+        """Return the Pauli error of a qubit pair."""
+        return {
+            pair: noise_utils.decay_constant_to_pauli_error(
+                noise_utils.xeb_fidelity_to_decay_constant(self.xeb_fidelity(*pair), num_qubits=2),
+                num_qubits=2,
+            )
+            for pair in self.all_qubit_pairs
+        }
+
+
+@dataclass(frozen=True)
+class CombinedXEBRBResult:
+    rb_result: ParallelRandomizedBenchmarkingResult
+    xeb_result: TwoQubitXEBResult
+
+    @property
+    def all_qubit_pairs(self) -> Sequence[Tuple['cirq.GridQubit', 'cirq.GridQubit']]:
+        return self.xeb_result.all_qubit_pairs
+
+    def _pauli_error(self, q0: 'cirq.GridQubit', q1: 'cirq.GridQubit') -> float:
+        """Return the inferred pauli error."""
+        q0, q1 = sorted([q0, q1])
+        single_q_paulis = self.rb_result.pauli_error()
+        return self.xeb_result.pauli_error()[(q0, q1)] - single_q_paulis[q0] - single_q_paulis[q1]
+
+    @cached_method
+    def pauli_error(self) -> Dict[Tuple['cirq.GridQubit', 'cirq.GridQubit'], float]:
+        """Return the inferred pauli error for all pairs."""
+        return {pair: self._pauli_error(*pair) for pair in self.all_qubit_pairs}
+
+    @cached_method
+    def decay_constant(self) -> Dict[Tuple['cirq.GridQubit', 'cirq.GridQubit'], float]:
+        """Return the inferred decay constant for all pairs."""
+        return {
+            pair: noise_utils.pauli_error_to_decay_constant(pauli, 2)
+            for pair, pauli in self.pauli_error().items()
+        }
+
+    @cached_method
+    def xeb_error(self) -> Dict[Tuple['cirq.GridQubit', 'cirq.GridQubit'], float]:
```

Comment:
Can we call this "inferred two-qubit Pauli error" or something?

Reply:
done